src/libos/src/process/process/mod.rs

```diff
@@ -123,12 +128,37 @@ impl Process {
     /// There are two reasons to force a process to exit:
     /// 1. Receiving a fatal signal;
     /// 2. Performing exit_group syscall.
+    /// 3. Performing execve syscall to terminate current process
     ///
     /// A process may be forced to exit many times, but only the first time counts.
     pub fn force_exit(&self, term_status: TermStatus) {
         self.forced_exit_status.force_exit(term_status);
     }
 
+    /// Difference with freeze defined in process:
+    /// - force_freeze only changes the thread status
+    /// - freeze element in process struct achieves the "freeze" behaviour
+    pub fn force_freeze(&self, tid: pid_t) {
+        self.forced_exit_status.force_freeze(tid);
+        let threads = self.threads();
+        for thread in threads.iter() {
+            if thread.tid() == tid {
```

Comment:
Do not get the point. Why do you freeze only one thread?

Reply:
Updated to freeze all threads including calling thread.